src/app/test-record/test-record.mapper.ts

```diff
@@ -1,131 +1,295 @@
+    testResultMapped.odometerReading = testResultFormData.odometerReading;
+    testResultMapped.odometerReadingUnits = testResultFormData.odometerReadingUnits;
+    testResultMapped.preparerName = !!testResultFormData.preparer
+      ? testResultFormData.preparer.split('(')[0]
+      : '';
+    testResultMapped.preparerId = !!testResultFormData.preparer
+      ? testResultFormData.preparer.match(/\((.*)\)/).pop()
+      : '';
+    testResultMapped.testStationName = !!testResultFormData.testStationNameNumber
+      ? testResultFormData.testStationNameNumber.split('(')[0]
+      : '';
+    testResultMapped.testStationPNumber = !!testResultFormData.testStationNameNumber
+      ? testResultFormData.testStationNameNumber.match(/\((.*)\)/).pop()
+      : '';
+    testResultMapped.testStationType = testResultFormData.testStationType;
+    testResultMapped.testerName = testResultFormData.testerName;
+    testResultMapped.testerEmailAddress = testResultFormData.testerEmailAddress;
+    testTypeMapped.seatbeltInstallationCheckDate =
+      testResultFormData.testType.seatbeltInstallationCheckDate;
+    testTypeMapped.numberOfSeatbeltsFitted = testResultFormData.testType.numberOfSeatbeltsFitted;
+    testTypeMapped.lastSeatbeltInstallationCheckDate =
+      testResultFormData.testType.lastSeatbeltInstallationCheckDate;
+    testTypeMapped.emissionStandard = testResultFormData.testType.emissionStandard;
+    testTypeMapped.smokeTestKLimitApplied = testResultFormData.testType.smokeTestKLimitApplied;
+    testTypeMapped.fuelType = testResultFormData.testType.fuelType;
+    if (!!testTypeMapped.modType) {
+      testTypeMapped.modType.code = !!testResultFormData.testType.modType
+        ? testResultFormData.testType.modType.split('-')[0]
+        : '';
+      testTypeMapped.modType.description = !!testResultFormData.testType.modType
+        ? testResultFormData.testType.modType.split('-')[1].trim()
+        : '';
+    }
+    testTypeMapped.modificationTypeUsed = testResultFormData.testType.modificationTypeUsed;
+    testTypeMapped.particulateTrapFitted = testResultFormData.testType.particulateTrapFitted;
+    testTypeMapped.particulateTrapSerialNumber =
+      testResultFormData.testType.particulateTrapSerialNumber;
+    testTypeMapped.additionalNotesRecorded = testResultFormData.testType.additionalNotesRecorded;
+
+    if (
+      !testTypesApplicable.seatBeltApplicable.hasOwnProperty(testTypeMapped.testTypeId) &&
+      !(testResultMapped.vehicleType === 'psv')
+    ) {
+      delete testTypeMapped.seatbeltInstallationCheckDate;
+      delete testTypeMapped.numberOfSeatbeltsFitted;
+      delete testTypeMapped.lastSeatbeltInstallationCheckDate;
+    }
+
+    if (
+      !testTypesApplicable.emissionDetailsApplicable.hasOwnProperty(testTypeMapped.testTypeId) &&
+      (!(testResultMapped.vehicleType === 'psv') ||
+        !(testResultMapped.vehicleType.valueOf() === 'hgv'))
+    ) {
+      delete testTypeMapped.emissionStandard;
```

Comment:
did you try using` testTypeMapped.emissionStandard = undefined` instead of delete. deleting is more expensive and not necessary I think

Reply:
Does not work:

Reply:
![image](https://user-images.githubusercontent.com/55185857/82810047-2ddd9200-9e96-11ea-916f-cfaf211079ce.png)

Reply:
![image](https://user-images.githubusercontent.com/55185857/82810081-4188f880-9e96-11ea-8d88-c3949f42e6f6.png)